Content.Server/GridPreloader/GridPreloaderSystem.cs

```diff
@@ -0,0 +1,114 @@
+using Content.Shared.CCVar;
+using Content.Shared.GridPreloader.Prototypes;
+using Content.Shared.GridPreloader.Systems;
+using Robust.Server.GameObjects;
+using Robust.Server.Maps;
+using Robust.Shared.Configuration;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Prototypes;
+using System.Linq;
+using System.Numerics;
+
+namespace Content.Server.GridPreloader;
+public sealed partial class GridPreloaderSystem : SharedGridPreloaderSystem
+{
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly MapLoaderSystem _map = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<GridPreloaderComponent, MapInitEvent>(OnMapInit);
+    }
+
+    private void OnMapInit(Entity<GridPreloaderComponent> preloader, ref MapInitEvent args)
+    {
+        if (!_cfg.GetCVar(CCVars.PreloadGrids))
+            return;
+
+        preloader.Comp.PreloadGridsMapId = _mapManager.CreateMap();
+        _mapManager.AddUninitializedMap(preloader.Comp.PreloadGridsMapId.Value);
+        _mapManager.SetMapPaused(preloader.Comp.PreloadGridsMapId.Value, true);
+
+        var mapUid = _mapManager.GetMapEntityId(preloader.Comp.PreloadGridsMapId.Value);
+        var globalXOffset = 0f;
+        foreach (var proto in _prototype.EnumeratePrototypes<PreloadedGridPrototype>())
+        {
+            for (int i = 0; i < proto.Copies; i++)
+            {
+                var options = new MapLoadOptions
+                {
+                    LoadMap = false,
+                };
+
+                // i dont use TryLoad, because he doesn't return EntityUid
+                var gridUid = _map.LoadGrid(preloader.Comp.PreloadGridsMapId.Value, proto.Path.ToString(), options);
+                if (!TryComp<MapGridComponent>(gridUid, out var mapGrid))
```

Comment:
Please don't add in obsolete code.

Reply:
How to get EntityUid from modern TryLoad()?

Reply:
You can never guarantee a single entity is returned from maploader. If you really want the functionality then make an engine PR and have it abort if there's more than 1 returned.

Reply:
Maybe we can get list<entityuid>?